src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleAssignmentService.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.roleassignment.RoleAssignmentRecord;
+import uk.gov.hmcts.ccd.data.roleassignment.RoleAssignmentRecordAttribute;
+import uk.gov.hmcts.ccd.data.roleassignment.RoleAssignmentRepository;
+import uk.gov.hmcts.ccd.data.roleassignment.RoleAssignmentResponse;
+import uk.gov.hmcts.ccd.domain.model.definition.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.definition.RoleAssignmentAttribute;
+import uk.gov.hmcts.ccd.domain.service.AccessControl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class RoleAssignmentService implements AccessControl {
+
+    private final RoleAssignmentRepository roleAssignmentRepository;
+
+    @Autowired
+    public RoleAssignmentService(RoleAssignmentRepository roleAssignmentRepository) {
+        this.roleAssignmentRepository = roleAssignmentRepository;
+    }
+
+    public List<RoleAssignment> getRoleAssignments(String userId) {
+        RoleAssignmentResponse roleAssignmentResponse = roleAssignmentRepository.getRoleAssignments(userId);
+        return mapToRoleAssignments(roleAssignmentResponse);
+    }
+
+    private List<RoleAssignment> mapToRoleAssignments(RoleAssignmentResponse roleAssignmentResponse) {
+        return roleAssignmentResponse.getRoleAssignmentRecords().stream()
+            .map(this::mapToRoleAssignment)
+            .collect(Collectors.toList());
+    }
+
+    private RoleAssignment mapToRoleAssignment(RoleAssignmentRecord roleAssignmentRecord) {
+        return RoleAssignment.builder()
```

Comment:
is it possible to create something like this for mapping, EntityToResponseDTOMapper

Reply:
I agree we should use mapstruct:
https://github.com/hmcts/ccd-data-store-api/blob/69ba3c92eadd918f3957fb0ce6fc207e9372abf2/src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/mapper/CaseDetailsMapper.java#L11
this is a lot of boilerplate code we could get rid of